packages/confirmation-modal/README.md

```diff
@@ -127,14 +127,27 @@ function ExampleComponent() {
+| Prop                | Type                      | Description                                                                                                                                           | Default     |
+| ------------------- | ------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
+| `open`              | `boolean`                 | Determines open state of `Modal` component                                                                                                            | `false`     |
+| `onConfirm`         | `function`                | Callback that fires when the action is confirmed. This can be used to set the modal to be closed.                                                     | `() => {}`  |
+| `onCancel`          | `function`                | Callback that fires when the cancel button, x button, or backdrop is clicked. This can be used to set the modal to be closed.                         | `() => {}`  |
+| `title`             | `string`                  | Title text to display above the main content text.                                                                                                    |             |
+| `children`          | `node`                    | Children that will be rendered inside `<ConfirmationModal />` component.                                                                              |             |
+| `buttonText`        | `string`                  | Text content of the confirmation button.                                                                                                              |             |
+| `variant`           | `'default'` \| `'danger'` | Sets the style variant.                                                                                                                               | `'default'` |
+| `requiredInputText` | `string`                  | If provided, a text prompt will be displayed and the confirmation button will be disabled until the text prompt is filled out with the required text. |             |
+| `className`         | `string`                  | Style to be applied to the container's root node.                                                                                                     |             |
+| ...                 | `ModalProps`              | Any other properties will be spread on the Modal element                                                                                              |             |
+
+##### ModalProps
+
+```typescript
+interface ModalProps {
+  size?: ModalSize;
+  setOpen?: (open: boolean) => void | React.Dispatch<SetStateAction<boolean>>;
+  shouldClose?: () => boolean;
+  closeOnBackdropClick?: boolean;
+  contentClassName?: string;
+}
+```
```

Comment:
these aren't explicitly supported by the confirmation or deletion modal

Reply:
No, but the rest parameters are spread directly on the modal, which is why I provided the documentation for the rest of the supported ModalProps

Reply:
The rest parameters being spread is an implementation detail. The only explicitly supported props are `open` and `className` which are both documented.

Reply:
So what do you think makes more sense in terms of typing the rest of the parameters that are spread on the Modal?

Reply:
They should be undocumented because they are not officially supported

Reply:
So I shouldn't acknowledge that there are props that can be spread on the root element?

Reply:
I think this is more about not documenting props that are exclusively handled through rest. It should be alright to acknowledge that we spread props that aren't explicitly documented on the root element.